src/documents/matching.py

```diff
@@ -1,27 +1,33 @@
 import logging
 import re
 from fnmatch import fnmatch
+from typing import Union
 
 from documents.classifier import DocumentClassifier
 from documents.data_models import ConsumableDocument
 from documents.data_models import DocumentSource
-from documents.models import ConsumptionTemplate
 from documents.models import Correspondent
 from documents.models import Document
 from documents.models import DocumentType
 from documents.models import MatchingModel
 from documents.models import StoragePath
 from documents.models import Tag
+from documents.models import Workflow
+from documents.models import WorkflowTrigger
 from documents.permissions import get_objects_for_user_owner_aware
 
 logger = logging.getLogger("paperless.matching")
 
 
 def log_reason(matching_model: MatchingModel, document: Document, reason: str):
     class_name = type(matching_model).__name__
+    name = (
```

Comment:
So this is not just the original matching model here anymore?  I don't actually find its usage changed in the diff

Reply:
Yea, this now accepts a MatchingModel or WorkflowTrigger. Triggers dont inherit the matching model because they dont have an owner or a name.

If I make the signature
`def log_reason(matching_model: Union[MatchingModel, WorkflowTrigger], document: Document, reason: str):`, would that be sufficient to address this in your mind?